src/entities/FormattingNode/index.ts

```diff
@@ -1,10 +1,21 @@
-import { TextNode } from '../TextNode';
-import { FormattingNodeConstructorParameters, FormattingNodeName, createFormattingNodeName } from './types';
+import {
+  FormattingNodeConstructorParameters,
+  FormattingNodeName,
+  FormattingNodeData
+} from './types';
+import { ChildNode, InlineFragment, InlineNode, InlineNodeSerialized, ParentNode } from '../interfaces';
+
+export * from './types';
+
+export interface FormattingNode extends ChildNode, ParentNode {
+}
 
 /**
  * FormattingNode class represents a node in a tree-like structure, used to store and manipulate formatted text content
  */
-export class FormattingNode {
+@ParentNode
+@ChildNode
+export class FormattingNode implements InlineNode {
```

Comment:
Why does this class implement only the InlineNode interface, but not ParentNode and ChildNode? I thought that you always need to implement interfaces to use methods from decorators.

Reply:
It doesn't need to implement ParentNode and ChildNode as those interfaces are implemented in mixins. We only need to declare that FormattingNode interface extends Parent and Child ones (on line 10)